### PR TITLE
DOCS-359 Removal of Intel Support

### PR DESCRIPTION
**Pull request by angelasimms**
Removing references to Intel Optane as no longer supported in 5.3.

### CODE REVIEW

File: docs/modules/storage/pages/high-density-memory.adoc [hunk 1]
```diff
@@ -166,106 +166,6 @@ To extend the memory available to the High-Density Memory Store, you can use per
...
-* `mode`: Defines the operational mode of the persistent memory device. Two modes are supported:
-** `MOUNTED`: If you choose this mode, the memory is mounted into the file system (aka FS DAX).
-** `SYSTEM_MEMORY`: If you choose this mode, Hazelcast uses the already onlined persistent memory portion of the system memory (aka KMEM DAX).
-* `directories`: A list of the mounted directories for the persistent memory device which are available for the storage of all data structures backed by the High-Density Memory Store. When you specify the mounted directories, the following rules apply:
-** Use of memory on the persistent memory device is enabled automatically, you do not need to explicitly
-set the `enabled` attribute to `true`
-** Set the `mode` to `MOUNTED`.
-+
-If you don't specify a directory for the persistent memory device, standard RAM is used instead.
-
```
Comment (blazember) [inline docs/modules/storage/pages/high-density-memory.adoc:268]: We should remove the below parts as well, until - including - "Performance Impacts on Different Use Cases". All below related to our Persistent Memory support.

**Review by blazember — CHANGES_REQUESTED** [review]
There are a few more section this PR should remove, commented on that.

Is there a higher-level task (JIRA, anything) for this that I can look/comment at? We still have pmem-related bits in the configuration and in the code as well, so after merging this, our manual won't cover an existing part of our functionality.

File: docs/modules/storage/pages/high-density-memory.adoc (hunk 1 above)
Comment (angelasimms) [reply docs/modules/storage/pages/high-density-memory.adoc:268]: Hi @blazember, I think I've removed all the relevant sections.

**Review by sakhouri21 — COMMENTED** [review]
AFAIK, currently there are no alternatives to Intel Optane Persistent Memory, so we cannot provide any alternatives. Even though this is no longer recommended for new customers, we should ensure that customers who are using Optane can continue to find configuration & troubleshooting  information.

**Comment by angelasimms** [comment]
> AFAIK, currently there are no alternatives to Intel Optane Persistent Memory, so we cannot provide any alternate recommendations. Even though this is no longer recommended for new customers, we should ensure that customers who are using Optane can continue to find configuration & troubleshooting information.

Hi @blazember and @sakhouri21, Based on your comments, I have reversed the changes (that removed relevant content) and shortened the admonition to say the following:

CAUTION: Intel(R) has discontinued support for Intel(R) Optane(TM) products. As a result, Hazelcast does not recommend using this configuration.

**Review by blazember — APPROVED** [review]
Approving as I understood too that this is the desired change our product needs. If there is a need for further changes, please re-request my review.

**Pull request merged**